rpm/packagescanner.go

```diff
@@ -34,6 +36,14 @@ var dbnames = map[string]struct{}{
 	"Packages": {},
 }
 
+// Database kinds
+const (
+	_ int = iota
+	backendBDB
+	backendNDB
+	backendSqlite
+)
```

Comment:
Fine if not, but might be nice to have a dedicated type and then this var is more self-documenting:

```go
oldDBs := make(map[string]backend, 0)
```

Reply:
The issue is that the old (bdb) backend is spread across multiple files. So the idea is to do a linear scan once of files that look like _any_ of the possible DB files and categorize them, then look at the ones that we're pretty sure are database and ignore extraneous files named `Packages` or the like.

Reply:
I guess this patch is also carved out from some hacking to attempt to support sqlite databases (which wouldn't require extracting an entire layer) so I think this "type" concept is left over